scripts/docsgen/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/coder/terraform-provider-coder/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// This script patches Markdown docs generated by `terraform-plugin-docs` to expose the original deprecation message.
+
+var (
+	reDeprecatedProperty = regexp.MustCompile("`([^`]+)` \\(([^,\\)]+), Deprecated\\) ([^\n]+)")
+)
+
+func main() {
+	p := provider.New()
+	err := exposeDeprecationMessage(p)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// FIXME flag to select dir
+}
+
+func exposeDeprecationMessage(p *schema.Provider) error {
+	// Patch data-sources
+	for dataSourceName, dataSource := range p.DataSourcesMap {
+		docFile := filepath.Join("docs", "data-sources", strings.Replace(dataSourceName, "coder_", "", 1)+".md")
+
+		err := adjustDocFile(docFile, dataSource.Schema)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Patch resources
+	for resourceName, resource := range p.ResourcesMap {
+		docFile := filepath.Join("docs", "resources", strings.Replace(resourceName, "coder_", "", 1)+".md")
+
+		err := adjustDocFile(docFile, resource.Schema)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Patch index
+	docFile := filepath.Join("docs", "index.md")
+	err := adjustDocFile(docFile, p.Schema)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func adjustDocFile(docPath string, schemas map[string]*schema.Schema) error {
+	doc, err := os.ReadFile(docPath)
+	if err != nil {
+		return err
+	}
+
+	result := writeDeprecationMessage(doc, schemas)
+
+	err = os.WriteFile(docPath, result, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeDeprecationMessage(doc []byte, schemas map[string]*schema.Schema) []byte {
+	return reDeprecatedProperty.ReplaceAllFunc(doc, func(m []byte) []byte {
+		matches := reDeprecatedProperty.FindSubmatch(m)
+		propertyName := matches[1]
+		description := matches[3]
+
+		sch := schemas[string(propertyName)]
+		if string(description) != sch.Description {
```

Comment:
nit: probably no harm to drop a log here for troubleshooting

Reply:
Fixed